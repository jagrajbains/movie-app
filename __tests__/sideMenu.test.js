import React from "react";
import {shallow} from "enzyme";
import renderer from "react-test-renderer";
import SideMenu from "../components/sideMenu";

describe("Testing the SideMenu component", () => {
    it("Should render the sideMenu component without any errors", () => {
        const component = renderer.create(<SideMenu />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Should show the App name as Movie DB", () => {
        const mockProps = "Movie DB";
        const wrapper = shallow(<SideMenu appName={mockProps}/>);
        expect(wrapper.find("h1").text()).toEqual("Movie DB");
    });
});