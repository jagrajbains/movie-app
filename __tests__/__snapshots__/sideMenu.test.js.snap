// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the SideMenu component Should render the sideMenu component without any errors 1`] = `
Array [
  <h1
    className="my-4"
  />,
  <div
    className="list-group"
  >
    <a
      className="list-group-item"
      href="#"
    >
      Category 1
    </a>
    <a
      className="list-group-item"
      href="#"
    >
      Category 2
    </a>
    <a
      className="list-group-item"
      href="#"
    >
      Category 3
    </a>
  </div>,
]
`;
